File: The kth Factor of n.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        if k < 1 or n < k or n > 1000:
            return -1

        if k == 1:
            return 1

        idx = 1
        for x in range(2, n + 1):
            if n % x == 0:
                idx += 1
            if idx == k:
                return x
        return -1

File: The kth Factor of n.c
int kthFactor(int n, int k) {
    if (k < 1 || n < k || n > 1000)
        return -1;

    if (k == 1)
        return 1;
    
    int index = 1;
    for (int i = 2; i <= n; ++i)
    {
        if (n % i == 0)
            ++index;
        if (index == k)
            return i;
    }
    return -1;
}